core/src/test/java/nu/yona/server/messaging/service/MessageServiceTest.java

```diff
@@ -0,0 +1,213 @@
+
+	@MockBean
+	private MessageDestinationRepository mockMessageDestinationRepository;
+
+	@MockBean
+	private MessageRepository mockMessageRepository;
+
+	@MockBean
+	private FirebaseService mockFirebaseService;
+
+	@Autowired
+	private MessageService service;
+
+	private static final String PASSWORD = "password";
+
+	private UUID userAnonId;
+	private UserAnonymized userAnonEntity;
+	private DeviceAnonymized deviceAnonEntity;
+
+	@Before
+	public void setUp()
+	{
+		when(mockUserService.generatePassword()).thenReturn("topSecret");
+
+		// Set up UserAnonymized instance.
+		MessageDestination anonMessageDestinationEntity = MessageDestination
+				.createInstance(PublicKeyUtil.generateKeyPair().getPublic());
+		Set<Goal> goals = new HashSet<>();
+		deviceAnonEntity = DeviceAnonymized.createInstance(0, OperatingSystem.ANDROID, "Unknown", 5,
+				Optional.of("Firebase-12345"));
+		deviceAnonymizedRepository.save(deviceAnonEntity);
+		userAnonEntity = UserAnonymized.createInstance(anonMessageDestinationEntity, goals);
+		userAnonEntity.addDeviceAnonymized(deviceAnonEntity);
+		UserAnonymizedDto userAnon = UserAnonymizedDto.createInstance(userAnonEntity);
+		userAnonId = userAnon.getId();
+
+		when(mockUserAnonymizedService.getUserAnonymized(userAnonId))
+				.thenReturn(UserAnonymizedDto.createInstance(userAnonEntity));
+
+		when(mockMessageDestinationRepository.findById(anonMessageDestinationEntity.getId()))
+				.thenReturn(Optional.of(anonMessageDestinationEntity));
+	}
+
+	@Override
+	protected Map<Class<?>, Repository<?, ?>> getRepositories()
+	{
+		Map<Class<?>, Repository<?, ?>> repositoriesMap = new HashMap<>();
+		repositoriesMap.put(MessageSource.class, mockMessageSourceRepository);
+		repositoriesMap.put(MessageDestination.class, mockMessageDestinationRepository);
+		repositoriesMap.put(Message.class, mockMessageRepository);
+		repositoriesMap.put(DeviceAnonymized.class, deviceAnonymizedRepository);
+		repositoriesMap.put(UserDevice.class, userDeviceRepository);
+		repositoriesMap.put(Goal.class, mockGoalRepository);
+		return repositoriesMap;
+	}
+
+	@Test
+	public void sendMessage_anonymousDestination_sendsFirebaseMessage()
+	{
+		Message message = Mockito.mock(Message.class);
+
+		service.sendMessage(message, UserAnonymizedDto.createInstance(userAnonEntity));
+
+		verify(mockFirebaseService, times(1)).sendMessage("Firebase-12345", message);
+	}
+
+	@Test
+	public void sendMessage_namedDestination_doesNotSendFirebaseMessage()
+	{
+		Message message = Mockito.mock(Message.class);
+
+		MessageDestination namedMessageDestination = Mockito.mock(MessageDestination.class);
+
+		User user = Mockito.mock(User.class);
+		when(user.getNamedMessageDestination()).thenReturn(namedMessageDestination);
+
+		service.sendNamedMessage(message, user);
+
+		verify(mockFirebaseService, never()).sendMessage("Firebase-12345", message);
+	}
+
+	@Test
+	public void prepareMessageCollection_transferringDirectMessagesToAnonymousDestination_sendsFirebaseMessage()
```

Comment:
I  already typed this as part of my previous comment:
> Are we sending them a Firebase notification when we move the messages from the named destination to the anonymous one?

Good to see this test answering the question!